server/aesdsocket.c

```diff
@@ -0,0 +1,289 @@
+	}
+
+	//handle all socket events
+	struct addrinfo addr_hints;
+	struct sockaddr_in clientaddr;
+	socklen_t sockaddrsize = sizeof(struct sockaddr_in); 
+    	ssize_t num_read=0;	//number of bytes read
+    	ssize_t num_write=0;	//number of bytes written
+
+	char read_data[BUFFER_SIZE];
+    	memset(read_data,0,BUFFER_SIZE);
+
+    	//clear the structure
+	memset(&addr_hints, 0, sizeof(addr_hints));
+	addr_hints.ai_family = SOCKET_FAMILY;
+	addr_hints.ai_socktype = SOCKET_TYPE;
+	addr_hints.ai_flags = SOCKET_FLAGS;
+
+	status_flag = getaddrinfo(NULL, PORT_NUM, &addr_hints, &new_addr_info);
+	if(status_flag != 0)
+	{
+		syslog(LOG_ERR, "Error in getaddrinfo with error code: %d\n", status_flag);	
+		return -1;
+	}
+	syslog(LOG_INFO, "getaddrinfo succeeded!\n");
+
+	//create a socket
+	socket_fd = socket(AF_INET6, SOCK_STREAM, 0);
+	if (socket_fd == -1)
+	{
+		syslog(LOG_ERR, "Socket creation failed.\n");
+		return -1;
+	}
+	syslog(LOG_INFO, "socket creation successful!\n");	
+
+	//bind
+	status_flag = bind(socket_fd, new_addr_info->ai_addr, new_addr_info->ai_addrlen);
+	if (status_flag == -1)
+	{
+		syslog(LOG_ERR, "Binding unsuccessful\n");
+		return -1;
+	}
+	syslog(LOG_INFO, "Binding successful\n");
+
+
+	//check if daemon is required, then run as daemon
+	if (isdaemon)
+	{	
+		status_flag = daemon(0,0);
+		if (status_flag == -1)
+		{
+			syslog(LOG_ERR, "Failed to run as Daemon\n");
+			return -1;
+		}
+	}
+
+	//create a file
+	int fd = creat(PATH, 0644);
+	if (fd == -1)
+	{
+		syslog(LOG_ERR, "File cannot be created");
+		return -1;
+	}
+	close(fd);
+	syslog(LOG_INFO, "File created!\n");	
+
+	while (true)
+	{
+        	buffer = (char *) malloc((sizeof(char)*BUFFER_SIZE));
+		if(buffer == NULL)
+		{
+			syslog(LOG_ERR, "Malloc failed!\n");
+			return -1;
+		}
+		syslog(LOG_INFO, "malloc succeded!\n");
+		
+		memset(buffer,0,BUFFER_SIZE);
+
+		//listen
+		status_flag = listen(socket_fd, 5);
+		if(status_flag == -1)
+		{
+			syslog(LOG_ERR, "Error Listening\n");
+			return -1;
+		}
+		syslog(LOG_INFO, "listen() succeeded!\n");
+
+		//accept
+		client_fd = accept (socket_fd, (struct sockaddr *)&clientaddr, &sockaddrsize);
+		if (client_fd == -1)
+		{
+			syslog(LOG_ERR, "Error accepting\n");
+			return -1;
+		}
+		//print the ip address of the connection
+		syslog(LOG_INFO, "accept succeeded! Accepted connection from: %s\n", inet_ntoa(clientaddr.sin_addr));
+
+		bool packet_rcvd = false;
+		int val;		
+		status_flag = 0;
+
+		while(!packet_rcvd)
+		{
+			num_read = recv(client_fd, read_data, BUFFER_SIZE, 0);		
+			if (num_read == -1)
+			{
+				syslog(LOG_ERR, "Receiving failed\n");
+				return -1;
+			}
+			else if (num_read == 0)
+			{
+				syslog(LOG_INFO, "recv succeeded\n");
+			}
+
+			for(val = 0;val < BUFFER_SIZE;val++)
+			{
+				if(read_data[val] == '\n')
+				{
+					packet_rcvd = true;		
+					val++;			
+					break;
+				}
+			}
+			packets_len += val;			
+			buffer = (char *) realloc(buffer,(size_t)(packets_len+1));
```

Comment:
@hwsingh27 what assumption is made here about the size of RAM heap relative to the total size of all packets sent?